### PR TITLE
Merchant repository respc

### PR DESCRIPTION
**Pull request by chsweet**
Type of Change:

- Refactor Merchant class
- Refactor MerchantRepository class and spec file

What does it implement:

- Updated Merchant class id variable to be and integer.
- Updated and refactored all tests for MerchantRepository class
- Refactored update method in MerchantRepository class to work

Any relevant logs, error outputs, etc.:

- All tests are passing

### CODE REVIEW

File: spec/merchant_repository_spec.rb [hunk 1]
```diff
@@ -11,60 +11,72 @@
...
 
-    xit 'can delete the merchant by id' do
-      @mock_repo.delete()
+    it 'can delete the merchant by id' do
+      @repo.delete(101)
 
-      expect(@merchant_repo.all_merchants).to eq([@merchant_1, @merchant_2, @merchant_4])
+      updated_all = []
+      @repo.all.each do |merchant|
+        updated_all_merchants << merchant.name
```
Comment (leehopper) [inline spec/merchant_repository_spec.rb]: I think this should be `updated_all  <<  merchant.name`, this is the only test that is failing for me and it looks like that is the issue.

File: lib/merchant_repository.rb [hunk 2]
```diff
@@ -15,48 +15,47 @@ def inspect
...
+    @all.max_by do |merchant|
       merchant.id
     end.id + 1
   end
 
   def create(attributes)
     new_merchant = Merchant.new({id: next_highest_merchant_id,
-                                 name: attributes})
-    @all_merchants << new_merchant
+                                 name: attributes}, self)
```
Comment (leehopper) [inline lib/merchant_repository.rb]: Thoughts on making this one line? With only 2 items in the hash, the current formatting catches my eye. Not essential though. If the `next_highest_merchant_id` method name was condensed that could help with the one line formatting.

**Review by leehopper — CHANGES_REQUESTED** [review]
Looks great to me, one tiny requested change (just a variable typo) and one formatting suggestion.

File: lib/merchant_repository.rb [hunk 3]
```diff
@@ -15,48 +15,46 @@ def inspect
...
-    @all_merchants.find do |merchant|
+    @all.find do |merchant|
       merchant.name.upcase == name.upcase
     end
   end
 
   def find_all_by_name(name)
-    @all_merchants.select do |merchant|
+    @all.select do |merchant|
         merchant.name.downcase.include?name.downcase
```
Comment (marlitas) [inline lib/merchant_repository.rb:36]: Nit-picky but if we're sticking with upcase all around we should change this to upcase too.

**Review by marlitas — COMMENTED** [review]
Just the on downcase to upcase thing. Everything else looks solid.

**Review by leehopper — APPROVED** [review]

**Pull request merged**